backend/rates/db.go

```diff
@@ -0,0 +1,64 @@
+package rates
+
+import (
+	"encoding/binary"
+	"math"
+	"path/filepath"
+	"time"
+
+	bbolt "github.com/coreos/bbolt"
+)
+
+func openRatesDB(dir string) (*bbolt.DB, error) {
+	opt := &bbolt.Options{Timeout: 5 * time.Second} // network disks may take long
+	return bbolt.Open(filepath.Join(dir, "rates.db"), 0600, opt)
+}
+
+// loadHistoryBucket loads data from an updater.historyDB bucket identified by the key.
+// The returned value is sorted by timestamp in ascending order.
+func (updater *RateUpdater) loadHistoryBucket(key string) ([]exchangeRate, error) {
+	var rates []exchangeRate
+	err := updater.historyDB.View(func(tx *bbolt.Tx) error {
+		bucket := tx.Bucket([]byte(key))
+		if bucket == nil {
+			return nil // no history exists for this key
+		}
+		return bucket.ForEach(func(k, v []byte) error {
+			timestamp := binary.LittleEndian.Uint64(k)
+			value := math.Float64frombits(binary.LittleEndian.Uint64(v))
+			rates = append(rates, exchangeRate{
+				value:     value,
+				timestamp: time.Unix(int64(timestamp), 0),
+			})
+			return nil
+		})
+	})
+	// Unfortunately, bbolt doesn't specify what kind of sorting order is used, if at all,
```

Comment:
Or maybe I should've used big endian in bolt db keys. But still, it's a shame I didn't find precise docs on the sorting order.

Reply:
https://github.com/etcd-io/bbolt#iterating-over-keys:

> Bolt stores its keys in byte-sorted order within a bucket. This makes sequential iteration over these keys extremely fast.

BigEndian should work I think.

Reply:
You could open an issue to clarify the wording in the docs. I am pretty sure the cursor iterates in the order it is stored, i.e. sorted. Many key-value databases do this and you get indexing for free if you encode the keys right.

This issue seems to confirm it, but it did not have a lot of discussion ;) https://github.com/etcd-io/bbolt/issues/101

Reply:
Yeah, I guess I was just complaining that one has to look at the readme and chase down issues. Instead, a single doc comment sentence in `ForEach` and `Cursor` would've been much more clear, committing to it so there are no breaking changes in the same major version. I've just sent them https://github.com/etcd-io/bbolt/pull/250.